File: analysis/import/SCMLogParser/src/test/java/de/maibornwolff/codecharta/importer/scmlogparser/parser/git/AuthorParserTest.java
package de.maibornwolff.codecharta.importer.scmlogparser.parser.git;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class AuthorParserTest {

    @Test
    public void parsesAuthorWithoutEmail() {
        String author = AuthorParser.parseAuthor("Author: TheAuthor");
        assertThat(author).isEqualTo("TheAuthor");
    }

    @Test
    public void parsesAuthorFromAuthorAndEmail() {
        String author = AuthorParser.parseAuthor("Author: TheAuthor <mail@example.com>");
        assertThat(author).isEqualTo("TheAuthor");
    }
}

File: analysis/import/SCMLogParser/src/test/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/parser/git/AuthorParserTest.kt
package de.maibornwolff.codecharta.importer.scmlogparser.parser.git

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class AuthorParserTest {

    @Test
    fun parsesAuthorWithoutEmail() {
        val author = AuthorParser.parseAuthor("Author: TheAuthor")
        assertThat(author).isEqualTo("TheAuthor")
    }

    @Test
    fun parsesAuthorFromAuthorAndEmail() {
        val author = AuthorParser.parseAuthor("Author: TheAuthor <mail@example.com>")
        assertThat(author).isEqualTo("TheAuthor")
    }
}